crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -4452,7 +4452,23 @@ impl<'a> Checker<'a> {
                 }
                 _ => false,
             } {
-                let (all_names, all_names_flags) = extract_all_names(&self.ctx, parent, current);
+                let (all_names, all_names_flags) = {
+                    let (mut names, flags) =
+                        extract_all_names(parent, |name| self.ctx.is_builtin(name));
+
+                    // Grab the existing bound __all__ values.
```

Comment:
This used to happen within `extract_all_names`, but it makes more sense to keep `extract_all_names` distinct from this existing-binding lookup.